src/Microsoft.AspNetCore.Mvc.Core/Internal/PhysicalFileResultExecutor.cs

```diff
@@ -44,15 +63,32 @@ public Task ExecuteAsync(ActionContext context, PhysicalFileResult result)
                     count: null,
                     cancellation: default(CancellationToken));
             }
-            else
+            else if (range == null || !fileStream.CanSeek)
             {
-                var fileStream = GetFileStream(result.FileName);
-
                 using (fileStream)
                 {
                     await fileStream.CopyToAsync(response.Body, DefaultBufferSize);
```

Comment:
Use StreamCopyOperation here too, you want the same cancellation logic if the writes start failing.